### DISCUSSION TITLE
Slide Styles/Layouts Rework/Enhancements

### DISCUSSION THREAD

**Discussion by abbasou** [ChurchApps/ChurchAppsSupport#312] [FreeShow] [answered]
As discussed at https://github.com/vassbo/freeshow/issues/132, theming slides will really need some added features. For starters, let me describe how themes/layouts/styles are used, at least in our case and I believe pretty broadly.

This goes in conjunction with NDI support, but is more directly related to changing the presentation of slide contents based on _specific outputs_. Here's how that looks:

1. There's an audience-facing screen (TV, projector, etc) that shows fullscreen slides including backgrounds, text, videos, etc.
2. There's a stage-facing screen that might need to show slide notes, or images, or potentially the same videos, etc.
3. There's a stream-facing output (NDI) that needs to output slide text (i.e. song lyrics), but as a lower third without motion backgrounds and with a transparent background color to overlay on top of a camera feed or whatever.

Critically, we need to be able to _change slide styles_ on a given output _during a "meeting"_. Sometimes we need lower thirds, sometimes we need full size videos or images, sometimes something else.

Short of simply copying the Propresenter method of doing this with "Looks" and slide "Actions", what are some ideas for implementing this critical feature in FreeShow? I'd also love to hear others' thoughts on how you all use your presentation software and various outputs. Looking forward to the continued development of FreeShow! It's looking great!

**Comment by vassbo** [comment, marked as answer]
Thanks for the great explanation. I will keep this in mind, and see what I can do about it.

I will prioritize optimizing the program, and fixing bugs first. So we will see when this will get done. And yes, I would love to get others opinion on how to do this in the best way possible.
